src/mlpack/core/util/using.hpp

```diff
@@ -75,6 +77,61 @@ namespace mlpack {
 
 #endif
 
+#if (ARMA_VERSION_MAJOR >= 12)
+  // By default, assume that we are using an Armadillo object.
+    template<typename MatType>
+    struct GetFillType
+    {
+      static const decltype(arma::fill::none) none;
+      static const decltype(arma::fill::zeros) zeros;
+      static const decltype(arma::fill::ones) ones;
+      static const decltype(arma::fill::randu) randu;
+      static const decltype(arma::fill::randn) randn;
+    };
+
+  #ifdef MLPACK_HAS_COOT
+    // If the matrix type is a Bandicoot type, use Bandicoot fill objects instead.
+    template<typename MatType,
+             typename = typename std::enable_if<is_coot_type<MatType>::value>::type*>
+    struct GetFillType
+    {
+      static const decltype(coot::fill::none) none;
+      static const decltype(coot::fill::zeros) zeros;
+      static const decltype(coot::fill::ones) ones;
+      static const decltype(coot::fill::randu) randu;
+      static const decltype(coot::fill::randn) randn;
+    };
+  #endif
+
+#else
```

Comment:
There's no difference in the `#else` here; but if I remember right, newer Armadillo versions use `constexpr` and older ones use `const`.  Can you make the newer version use `constexpr`?

Reply:
oh, I just removed all of that, I do not think I have pushed the modification yet
I only left one case, for all, I think when we make the base version is arma 12, then we can use `constexpr`

Reply:
If it works with `const`, that's fine with me, although it would be worth adding a comment to indicate the minimum required Armadillo version to switch to `constexpr` (I think that is in some other comment in this PR somewhere). :+1: